test/test_jit.py

```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
-            return math.log(x)
-
-        def test_log_float(x):
-            # type: (float) -> float
-            return math.log(x)
-
-        def test_log_base_float(x, y):
-            # type: (float, float) -> float
-            return math.log(x, y)
-
-        def test_log1p_int(x):
-            # type: (int) -> float
-            return math.log1p(x)
-
-        def test_log1p_float(x):
-            # type: (float) -> float
-            return math.log1p(x)
-
-        def test_log10_int(x):
-            # type: (int) -> float
-            return math.log10(x)
-
-        def test_log10_float(x):
-            # type: (float) -> float
-            return math.log10(x)
+        inf = float("inf")
+        NaN = float("nan")
+        mx_int = 2147483647
+        mn_int = -2147483647 - 1
+        float_vals = [inf, NaN, 0.0, 1.0, 2.2, -1.0, -0.0, -2.2, -inf, 1, 0, 2]
+        int_vals = list(range(-5, 5, 1)) + [mx_int + 5, mx_int * 2, mn_int - 5, mn_int * 2]
 
-        def test_exp_int(x):
-            # type: (int) -> float
-            return math.exp(x)
+        def checkMath(func_name, num_args, is_float=True, ret_type="float", debug=False, vals=None, args_type=None):
+            funcs_template = dedent('''
+            def func(a, b):
+                # type: {args_type} -> {ret_type}
+                return math.{func}({args})
+            ''')
+            if num_args == 1:
+                args = "a"
+            elif num_args == 2:
+                args = "a, b"
+            else:
+                raise RuntimeError("Test doesn't support more than 2 arguments")
+            if args_type is None:
+                args_type = "(float, float)" if is_float else "(int, int)"
+            funcs_str = funcs_template.format(func=func_name, args=args, args_type=args_type, ret_type=ret_type)
+            scope = {}
+            execWrapper(funcs_str, globals(), scope)
+            cu = torch.jit.CompilationUnit(funcs_str)
+            fs = cu.func
+            f = scope['func']
 
-        def test_exp_float(x):
-            # type: (float) -> float
-            return math.exp(x)
 
-        def test_sqrt_int(x):
-            # type: (int) -> float
-            return math.sqrt(x)
+            if vals is None:
+                vals = float_vals if is_float else int_vals
+                vals = [(i, j) for i in vals for j in vals]
 
-        def test_sqrt_float(x):
-            # type: (float) -> float
-            return math.sqrt(x)
+            for a, b in vals:
+                resf = None
+                resfs = None
+                try:
+                    resf = f(a, b)
+                except Exception as e:
+                    resf = e
+                try:
+                    resfs = fs(a, b)
+                except Exception as e:
+                    resfs = e
+                if debug:
+                    print("in: ", a, b)
+                    print("out: ", resf, resfs)
+                if resf != resfs:
```

Comment:
I can't use `assertEqual` here because there are some specific conditions (`nan` should be equal, we want to silently ignore errors, etc.)

Actually, lemme document that.